### DISCUSSION TITLE
Default Mainnet RPC Issues - Experiencing Downtime or Heavy Rate Limiting

### DISCUSSION THREAD

**Discussion by highnet** [wevm/viem#3155] [Question]
Hi viem developers and viem users,

I've been encountering issues with the default Mainnet RPC in viem. It seems to be either down or heavily rate-limited. My [dAPP](https://app.sandclock.org/), which had been running smoothly before, started failing with errors related to RPC availability. Switching to another public RPC provider resolved the issue.

Here’s what I was doing with the default setup:

```typescript
import { createPublicClient, http } from 'viem';
import { mainnet } from 'viem/chains';

const client = createPublicClient({
  chain: mainnet,
  transport: http(),
});

// Example query
async function getBlockNumber() {
  const blockNumber = await client.getBlockNumber();
  console.log('Current block number:', blockNumber);
}

getBlockNumber();
```

With the default Mainnet RPC, I noticed frequent failures, and I suspect it's due to rate-limiting or downtime on the public endpoint. After switching to another public RPC (e.g.,Ankr), everything worked fine:

```typescript
const client = createPublicClient({
  chain: mainnet,
  transport: http('https://rpc.ankr.com/eth'),
});
```
This immediately resolved the issue.

**Questions for the Community**
- Is anyone else experiencing similar issues with the default Mainnet RPC in viem?
What’s the actual default RPC endpoint being used for Mainnet? I couldn’t find this clearly documented.
- Do you recommend sticking with the default RPC for production applications, or should we always use a custom RPC (e.g., Ankr)?
- Any advice or recommendations on handling RPC reliability for viem-based applications would be greatly appreciated.

Thanks in advance for your help!

**Comment by tmm** [comment]
When you don't provide your own RPC URL (e.g. `http()`), Viem uses the default public RPC URL. These are shared resources so they can be okay for prototyping, but should not be used for production applications.

**Comment by highnet** [comment]
Thanks for the quick response.
